### ISSUE TITLE
async support

### ISSUE THREAD

**Issue by realcr** [untitaker/rust-atomicwrites#38]
Hi, thank you for this project!
I would like to use `rust-atomicwrites` in an async environment. I was wondering if we could have an async implementation for the part which writes into the file.

My current implementation uses a thread pool around the atomic write part, it looks somewhat like this:

```rust
        let mutate_fut = future::lazy(move |_| {
            atomic_db
                .mutate_db(&mutations[..])
                .map_err(DatabaseError::AtomicDbError)?;
            Ok(atomic_db)
        });
        let handle = database_spawner
            .spawn_with_handle(mutate_fut)
            .map_err(|_| DatabaseError::SpawnError)?;

        atomic_db = handle.await?;
```

In the code above, `mutate_db()` internally uses `rust-tomicwrites` to write to disk. Note that I am using an extra thread pool (Having just one thread) represented by `database_spawner`, to make sure I don't block other async operations in my main thread pool.

async has recently landed on stable, and we have `async-std` that should allow dealing with files.
I am not sure about what could be the atomicity related challenges in having such implementation. I heard some claims that async file access is not real in some operation systems, so maybe mixing it with atomicity requirement might be too much.

What is your opinion?

**Comment by untitaker** [comment]
Yeah async file access is not quite there yet. Sentry has a few old-style futures services and we just spawn all disk IO into a thread pool because it boils down to the same syscalls.

**Comment by untitaker** [comment]
Fwiw in any case I want to sit the async competition in Rust out. Unclear if Tokio or async-std is the way forward.
